hathor/sysctl/runner.py

```diff
@@ -13,13 +13,20 @@
 # limitations under the License.
 
 import json
+import re
 from typing import TYPE_CHECKING, Any
 
 from hathor.sysctl.exception import SysctlRunnerException
 
 if TYPE_CHECKING:
     from hathor.sysctl.sysctl import Sysctl
 
+# - It starts with an opening square bracket [.
+# - It ends with a closing square bracket ].
+# - The elements are separated by commas and can be followed by optional whitespace.
+# - There can be zero or more elements in the array (an empty array is allowed).
+array_pattern = r'^\s*\[\s*(?:[^\[\],]+(?:\s*,\s*[^\[\],]+)*)?\s*\]\s*$'
```

Comment:
The regex pattern is hard to understand for me. I understand the "positive" cases, but what are the "negative" cases: what are we trying to prevent from being parsed as a list/array by using this regex?

Reply:
I've update it and tried to make the regex clearer.